drivers/usbhost/usbhost_hub.c

```diff
@@ -965,12 +965,34 @@ static void usbhost_hub_event(FAR void *arg)
               if (connport->devclass != NULL)
                 {
                   CLASS_DISCONNECTED(connport->devclass);
-                  connport->devclass = NULL;
-                }
 
-              /* Free any resources used by the hub port */
+                  if (connport->devclass->connect == usbhost_connect)
```

Comment:
This is how I check for a hub. Only the hub class will set connect to the usbhost_connect function in this file.